src/acl/RegexData.cc

```diff
@@ -264,9 +242,19 @@ ACLRegexData::parse()
         }
     }
 
-    if (!compileOptimisedREs(data, sl, flagsAtLineStart)) {
-        debugs(28, DBG_IMPORTANT, "WARNING: optimisation of regular expressions failed; using fallback method without optimisation");
+    try {
+        // ignore the danger of merging invalid REs into a valid "optimized" RE
```

Comment:
This source code comment applies to the old/official code as well -- this PR does not affect the existing tradeoff between configuration speed and correctness: We (still) choose to ignore the fact that broken REs can be "optimized" into a completely different[^1] valid combined RE:

    compileOptimisedREs: adding RE 're)?((1'
    compileOptimisedREs: adding RE '2))+(re'
    RegexPattern: (re)?((1)|(2))+(re)
    compileOptimisedREs: 2 REs are optimised into one RE.

[^1]: It is impossible to meaningfully compare with broken REs, but we can assume that the combined version was _not_ the REs author intent!